plugin/trino-exasol/src/test/java/io/trino/plugin/exasol/ExasolTpchTables.java

```diff
@@ -0,0 +1,111 @@
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static io.trino.plugin.exasol.TestingExasolServer.TEST_SCHEMA;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+final class ExasolTpchTables
+{
+    private static final Logger log = Logger.get(ExasolTpchTables.class);
+
+    private ExasolTpchTables() {}
+
+    static void copyAndIngestTpchData(MaterializedResult rows, TestingExasolServer server, String tableName)
+    {
+        Path tempFile = createCsvFile(rows, tableName);
+        StringBuilder columnDefinitions = new StringBuilder();
+        for (int i = 0; i < rows.getTypes().size(); i++) {
+            if (i != 0) {
+                columnDefinitions.append(", ");
+            }
+            columnDefinitions.append(rows.getColumnNames().get(i) + " " + convertType(tableName, rows.getColumnNames().get(i), rows.getTypes().get(i)));
+        }
+        @Language("SQL")
+        String createTableStatement = format("CREATE TABLE %s.%s (%s)", TEST_SCHEMA, tableName, columnDefinitions);
+        log.info("Creating table %s using definition '%s'", tableName, createTableStatement);
+        server.execute(createTableStatement);
+        @Language("SQL")
+        String importStatement = format("""
+                IMPORT INTO %s.%s
+                FROM LOCAL CSV FILE '%s'
+                ROW SEPARATOR = 'LF'
+                COLUMN SEPARATOR = ','""", TEST_SCHEMA, tableName, tempFile.toAbsolutePath());
+        server.execute(importStatement);
+    }
+
+    private static String convertType(String tableName, String columnName, Type type)
+    {
+        if (tableName.equals(TpchTable.ORDERS.getTableName())) {
+            if (("o_" + columnName).equals(OrderColumn.TOTAL_PRICE.getColumnName())) {
+                // Column type is double but tests expect precision 2
+                return "decimal(10,2)";
+            }
+        }
+        return switch (type.getDisplayName()) {
+            case "bigint" -> // Exasol interprets bigint as decimal(36,0), but tests expect decimal(19,0)
+                    "decimal(19,0)";
+            case "integer" -> "decimal(10,0)";
+            default -> type.getDisplayName();
+        };
+    }
+
+    private static Path createCsvFile(MaterializedResult rows, String tableName)
+    {
+        try {
+            Path tempFile = Files.createTempFile("exasol_" + tableName, ".csv");
+            writeDataAsCsv(rows, tempFile);
+            return tempFile;
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private static void writeDataAsCsv(MaterializedResult rows, Path dataFile)
+            throws IOException
+    {
+        try (BufferedWriter bw = Files.newBufferedWriter(dataFile, UTF_8)) {
```

Comment:
Rename to `bufferedWriter`. We avoid abbreviations. https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#avoid-abbreviations